src/seeds.proposals.cpp

```diff
@@ -261,237 +261,369 @@ bool proposals::is_active(name account, uint64_t cutoff_date) {
+    permission_level(get_self(), "active"_n),
+    get_self(),
+    "evalproposal"_n,
+    std::make_tuple(proposal_id, prop_cycle)
+  );
+  // trx.delay_sec = 1;
+  trx.send(proposal_id, _self);
+}
 
-    update_cycle();
-    update_cycle_stats(active_props, eval_props);
-    updatevoices();
-    
-    transaction trx_erase_participants{};
-    trx_erase_participants.actions.emplace_back(
-      permission_level(_self, "active"_n),
-      _self,
-      "erasepartpts"_n,
-      std::make_tuple(number_active_proposals)
-    );
-    // I don't know how long delay I should use
-    // trx_erase_participants.delay_sec = 5;
-    trx_erase_participants.send(eosio::current_time_point().sec_since_epoch(), _self);
+void proposals::send_update_voices () {
+  transaction trx{};
+  trx.actions.emplace_back(
+    permission_level(_self, "active"_n),
+    _self,
+    "updatevoice"_n,
+    std::make_tuple(uint64_t(0))
+  );
+  // trx.delay_sec = 1;
+  trx.send(eosio::current_time_point().sec_since_epoch() + contracts::proposals.value, _self);
 }
 
-void proposals::testperiod() {
-    require_auth(_self);
+void proposals::onperiod() {
+  require_auth(get_self());
 
-    auto props_by_status = props.get_index<"bystatus"_n>();
-    uint64_t prop_majority = config_get(name("propmajority"));
-    uint64_t number_active_proposals = get_size(prop_active_size);
-    uint64_t total_eligible_voters = get_size(user_active_size);
-    check(total_eligible_voters > 0, "no eligible voters - likely an error; can't run proposals.");
-    
-    uint64_t quorum =  get_quorum(number_active_proposals);
+  cycle_table c = cycle.get_or_create(get_self(), cycle_table());
 
-    std::vector<uint64_t> active_props;
-    std::vector<uint64_t> eval_props;
+  auto citr = cyclestats.find(c.propcycle);
+  if (citr != cyclestats.end()) {
+    cyclestats.modify(citr, _self, [&](auto & item){
+      item.total_eligible_voters = get_size(user_active_size);
+    });
+  }
 
-    // TODO this is not working at the moment, use old way... FIX after this cycle.
+  uint64_t number_active_proposals = get_size(prop_active_size);
 
-    // find smallesd prop id that's in open or eval stage
-    // this way we skip all proposals that are definitely already passed or rejected
-    // auto pps_itr = props_by_status.begin();
-    // uint64_t smallest_prop_id = 0;
-    // while (pps_itr != props_by_status.end() && 
-    //   (pps_itr -> status == status_open || pps_itr -> status == status_evaluate) ) {
-    //   smallest_prop_id  = std::min(smallest_prop_id, pps_itr -> id);
-    //   pps_itr++;
-    // }
+  auto props_by_stage = props.get_index<"bystage"_n>();
 
-    // print("smallest id: "+std::to_string(smallest_prop_id) );
-      
-    // auto pitr = props.find(smallest_prop_id);
-    auto pitr = props.find(50);
+  auto spitr = props_by_stage.find(stage_staged.value);
+  while (spitr != props_by_stage.end() && spitr->stage == stage_staged) {
+    send_eval_prop(spitr->id, c.propcycle);
```

Comment:
I would create a separate method for eval

Reply:
In any case we can put that in another PR - maybe leave this as is, one change at a time!

Reply:
I agree with you, another function for just staged proposals is a good idea, but I also think that could go into another pull request